Runs: update semantic colors (now supports config key-value pairs) (#1420)
wandb/docs: content/en/guides/models/track/runs/color-code-runs.md

title: Semantic run plot legends
---

> Create meaningful legends for line plots to visually distinguish them based on a metric or run configuration you log to W&B. Identify trends and patterns based on the highest, lowest, or latest values of a metric key-value or run configuration's key-value pairs across your training and evaluation runs. Runs are bucketed and colored based on the plot's defined metric key-value or run configuration key-value pair, Y value, and number of buckets. 

Reviewer: This could use some copy massaging: "...values of a metric key-value or run configuration's key-value pairs..."

Navigate to your workspace's settings page to configure metric or configuration-based colors for runs:

1. Navigate to your W&B project.
2. Select the **Workspace** tab from the project sidebar.
3. Click on the **Settings** icon (⚙️) in the top right corner.
4. From the drawer, select **Runs** then select **Key-based colors**.
    - From the **Key** dropdown, select the metric you want to use for assigning colors to runs.
    - From the **Y value** dropdown, select the y value you want to use for assigning colors to runs.
    - Set the the number of buckets to a value from 2 to 8.

The following sections describe how to set the metric and y value and as how to customize the buckets used for assigning colors to runs.

## Set a metric

> The metric options in your **Key** dropdown are derived from the key-value pairs [you log to W&B]({{< relref "guides/models/track/runs/color-code-runs/#custom-metrics" >}}) and [default metrics]({{< relref "guides/models/track/runs/color-code-runs/#default-metrics" >}}) defined by W&B.

Reviewer: ```suggestion
The metric options in your **Key** dropdown are derived from the key-value pairs [you log to W&B]({{< relref "guides/models/track/runs/color-code-runs/#custom-metrics" >}}) and W&B's [default metrics]({{< relref "guides/models/track/runs/color-code-runs/#default-metrics" >}}).
```

## Customize buckets

Buckets are ranges of values that W&B uses to categorize runs based on the metric or configuration key you select. Buckets are evenly distributed across the range of values for the specified metric or configuration key and each bucket is assigned a unique color. Runs that fall within that bucket's range are displayed in that color. 

Consider the following:

{{< img src="/images/track/color-coding-runs.png" alt="" >}}

- **Key** is set to `"Accuracy"` (abbreviated as `"acc"`).
- **Y value** is set to `"Max"`

> With this set up, W&B colors run based on their accuracy values, using the maximum accuracy value logged for each run. 

Reviewer: ```suggestion
With this configuration, W&B colors each run based on the maximum accuracy values it logged.. 
```

- **Key** is set to `"Accuracy"` (abbreviated as `"acc"`).
- **Y value** is set to `"Max"`

> With this configuration, W&B colors each run based on their accuracy values. The colors vary from a light yellow color to a deep color. Lighter colors represent lower accuracy values, while deeper colors represent higher accuracy values.

Reviewer: ```suggestion
With this configuration, W&B colors each run based on its accuracy values. The colors vary from a light yellow color to a deep color. Lighter colors represent lower accuracy values, while deeper colors represent higher accuracy values.
```
